internal/impl/aws/processor_bedrock.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed as a Redpanda Enterprise file under the Redpanda Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+// https://github.com/redpanda-data/connect/blob/main/licenses/rcl.md
+
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"unicode/utf8"
+
+	"github.com/aws/aws-sdk-go-v2/service/bedrockruntime"
+	bedrocktypes "github.com/aws/aws-sdk-go-v2/service/bedrockruntime/types"
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"github.com/redpanda-data/connect/v4/internal/impl/aws/config"
+)
+
+const (
+	bedFieldModel        = "model"
+	bedFieldUserPrompt   = "prompt"
+	bedFieldSystemPrompt = "system_prompt"
+	bedFieldMaxTokens    = "max_tokens"
+	bedFieldTemp         = "stop"
+	bedFieldStop         = "temperature"
+	bedFieldTopP         = "top_p"
+)
+
+func init() {
+	err := service.RegisterProcessor("aws_bedrock_chat", newBedrockConfigSpec(), newBedrockProcessor)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func newBedrockConfigSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Summary("Generates responses to messages in a chat conversation, using the AWS Bedrock API.").
+		Description(`This processor sends prompts to your chosen large language model (LLM) and generates text from the responses, using the AWS Bedrock API.
+For more information, see the https://docs.aws.amazon.com/bedrock/latest/userguide[AWS Bedrock documentation^].`).
+		Categories("AI").
+		Version("4.34.0").
+		Fields(config.SessionFields()...).
+		Field(service.NewStringField(bedFieldModel).
+			Examples("amazon.titan-text-express-v1", "anthropic.claude-3-5-sonnet-20240620-v1:0", "cohere.command-text-v14", "meta.llama3-1-70b-instruct-v1:0", "mistral.mistral-large-2402-v1:0").
+			Description("The model ID to use. For a full list see the https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids.html[AWS Bedrock documentation^].")).
+		Field(service.NewStringField(bedFieldUserPrompt).
+			Description("The prompt you want to generate a response for. By default, the processor submits the entire payload as a string.").
+			Optional()).
+		Field(service.NewStringField(bedFieldSystemPrompt).
+			Optional().
+			Description("The system prompt to submit to the AWS Bedrock LLM.")).
+		Field(service.NewIntField(bedFieldMaxTokens).
```

Comment:
I think this has to be at least 1: https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_InferenceConfiguration.html#bedrock-Type-runtime_InferenceConfiguration-maxTokens

Reply:
Oh nice catch - I didn't even think someone would try that. Added a lint rule